packages/devp2p/test/integration/rlpx-simulator.spec.ts

```diff
@@ -6,94 +6,100 @@ import { DISCONNECT_REASON } from '../../src/types.js'
-      await util.delay(500)
-      util.destroyRLPXs(rlpxs)
+    const { rlpxs, peer } = util.initTwoPeerRLPXSetup(undefined, undefined, undefined, basePort + 1)
+    rlpxs[0]['_dpt']!.addPeer(peer).catch(() => {
+      throw new Error('Peering failed')
+    })
+    await new Promise((resolve) => {
+      rlpxs[0].events.on('peer:added', async (peer) => {
+        assert.equal(
+          peer['_port'],
+          basePort + 1,
+          'should have added peer on peer:added after successful handshake',
+        )
+        assert.equal(rlpxs[0].getPeers().length, 1, 'peer list length should be 1')
+        assert.equal(rlpxs[0]._getOpenSlots(), 9, 'should have maxPeers - 1 open slots left')
+        await util.delay(500)
+        util.destroyRLPXs(rlpxs)
+        resolve(undefined)
+      })
     })
   })
-
-  it('RLPX: ban node with missing tcp port', () => {
-    const rlpxs = util.initTwoPeerRLPXSetup(undefined, undefined, undefined, 40444)
-
-    rlpxs[0].events.on('peer:added', async () => {
-      const peer = {
-        id: hexToBytes('0xabcd'),
-        address: '127.0.0.1',
-        udpPort: 30308,
-        tcpPort: null,
-      }
-      assert.notOk(
-        rlpxs[0]['_dpt']!['_banlist'].has(peer),
-        'should not be in ban list before bad peer discovered',
-      )
-      rlpxs[0]['_dpt']!.events.emit('peer:new', peer)
-      assert.ok(
-        rlpxs[0]['_dpt']!['_banlist'].has(peer),
-        'should be in ban list after bad peer discovered',
-      )
-      await util.delay(500)
-      util.destroyRLPXs(rlpxs)
+  it('RLPX: ban node with missing tcp port', async () => {
+    const { rlpxs, peer } = util.initTwoPeerRLPXSetup(undefined, undefined, undefined, 40444)
+    rlpxs[0]['_dpt']!.addPeer(peer).catch(() => {
+      throw new Error('Peering failed')
+    })
+    await new Promise((resolve) => {
+      rlpxs[0].events.on('peer:added', async () => {
+        const peer = {
+          id: hexToBytes('0xabcd'),
+          address: '127.0.0.1',
+          udpPort: 30308,
+          tcpPort: null,
+        }
+        assert.notOk(
+          rlpxs[0]['_dpt']!['_banlist'].has(peer),
+          'should not be in ban list before bad peer discovered',
+        )
+        rlpxs[0]['_dpt']!.events.emit('peer:new', peer)
+        assert.ok(
+          rlpxs[0]['_dpt']!['_banlist'].has(peer),
+          'should be in ban list after bad peer discovered',
+        )
+        await util.delay(500)
+        util.destroyRLPXs(rlpxs)
+        resolve(undefined)
+      })
     })
   })
-
-  it('RLPX: remove node', () => {
-    const rlpxs = util.initTwoPeerRLPXSetup(undefined, undefined, undefined, 40504)
-
-    try {
-      rlpxs[0].events.once('peer:added', (peer) => {
-        rlpxs[0].disconnect(peer['_remoteId'])
+  it('RLPX: remove node', async () => {
+    const { rlpxs, peer } = util.initTwoPeerRLPXSetup(undefined, undefined, undefined, 40504)
+    rlpxs[0]
+      ['_dpt']!.addPeer(peer)
```

Comment:
The reason I moved the `addPeer` call out of `initTwoPeerRLPXSetup` is to be able to chain the disconnect here before checking for `peer:removed` events in the awaited promise that follows.